packages/app/pages/api/collaboration/models/[modelName]/documents/[documentTitle]/index.ts

```diff
@@ -0,0 +1,74 @@
+import { getLoggedInUser } from 'auth/user'
+import {
+    getDocumentCollaborators,
+    setDocumentCollaborator,
+} from 'collaboration/service'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { respondAsJson } from 'utils/api'
+import { apiError, ErrorCode, HttpException } from 'utils/errors'
+import { safeParseJSON } from 'utils/json'
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    const modelName = decodeURIComponent(req.query.modelName.toString())
+    const documentTitle = decodeURIComponent(req.query.documentTitle.toString())
+    const user = await getLoggedInUser(req, res)
+
+    switch (req.method) {
+        case 'GET': {
+            // Unlike most GET endpoints, unauthenticated users should not have access to an authenicated user's actions.
```

Comment:
This is worth noting. Also, I'm noodling over how to differentiate between `userCanAccessModel` and these `!user` checks. I feel like we could rename those functions to be clearer about their restrictions.